model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+            String entryId = entry.getKey();
+            int parsedEntryId;
+            try {
+                parsedEntryId = Integer.parseInt(entryId);
+            }catch (Exception e){
+                continue;
+            }
+            //If the Teltonika Parameter HashMap doesn't contain the requested AVL ID, continue:
+
+            if (!params.containsKey(parsedEntryId)) continue;
+
+            //Retrieve the parameter data
+            TeltonikaParameter parameter = params.get(Integer.valueOf(entry.getKey()));
+
+            //Create the MetaItem Map
+            MetaMap metaMap = new MetaMap();
+
+            // Figure out its attributeType
+            ValueDescriptor<?> attributeType = GetAttributeType(parameter);
+
+            //Retrieve its coerced value
+            Optional<?> value;
+            try {
+                //Inner method returns Optional.empty, but still throws and prints exception. A lot of clutter, but the exception is handled.
+                value = ValueUtil.getValueCoerced(entry.getValue(), attributeType.getType());
+                if (value.isEmpty()){
+                    attributeType = ValueType.TEXT;
+                    value = Optional.of(entry.getValue().toString());
+                }
+            } catch (Exception e){
+                value = Optional.of(entry.getValue().toString());
+                attributeType = ValueType.TEXT;
+                logger.severe("Failed value parse");
+                logger.severe(e.toString());
+            }
+            Optional<?> originalValue = value;
+
+            double multiplier = 1L;
+            //If value was parsed correctly,
+            // Multiply the value with its multiplier if given
+            if(!Objects.equals(parameter.multiplier, "-")){
+                Optional<?> optionalMultiplier = ValueUtil.parse(parameter.multiplier, attributeType.getType());
+
+                if(optionalMultiplier.isPresent()){
+
+                    if(!ValueUtil.objectsEquals(optionalMultiplier.get(), multiplier)) multiplier = (Double) optionalMultiplier.get();
+                    try{
+                        double valueNumber = (double) value.get();
+
+                        value = Optional.of(valueNumber * multiplier);
+                        //If the original value is unequal to the new (multiplied) value, then we have to also multiply the constraints
+
+                    }catch (Exception e){
+                        logger.severe(parameter.propertyName + "Failed multiplier");
+                        logger.severe(e.toString());
+                        throw e;
+                    }
+                }
+
+                //TODO: Fix frontend to be able to display even custom units, not just the predefined ones
+                //possibly prepend the unit with the string "custom."? So that it matches the predefined format
+                //Add on its units
+                if(!Objects.equals(parameter.units, "-")){
```

Comment:
`parameter.units` could be null or empty?

Reply:
Since these are the values parsed from the Teltonika wiki, all of the no-values have that dash as a value. Should I do some pre-processing when importing the values to set them to proper values, and then check those values in this method?